File: crop_image.py
# pip3 install Pillow
# Crop image into a square

from PIL import Image

def crop_to_square(input_path, output_path):
    # Open the image
    img = Image.open(input_path)

    # Find the center of the image
    center_x, center_y = img.width // 2, img.height // 2

    # Calculate the crop box
    size = min(img.width, img.height)
    left = center_x - size // 2
    top = center_y - size // 2
    right = center_x + size // 2
    bottom = center_y + size // 2

    # Crop the image
    img_cropped = img.crop((left, top, right, bottom))

    # Save the cropped image
    img_cropped.save(output_path)

# Example usage:
input_image_path = 'path/to/your/input/image.jpg'
output_image_path = 'path/to/your/output/cropped_image.jpg'
crop_to_square(input_image_path, output_image_path)

File: crop_image.js
function cropToSquare(inputImage, outputCanvas) {
    const img = new Image();
    img.src = inputImage;

    img.onload = function () {
        const canvas = document.createElement('canvas');
        const size = Math.min(img.width, img.height);
        canvas.width = size;
        canvas.height = size;

        const ctx = canvas.getContext('2d');
        ctx.drawImage(
            img,
            (img.width - size) / 2,
            (img.height - size) / 2,
            size,
            size,
            0,
            0,
            size,
            size
        );

        // Save or display the cropped image
        // For example, document.body.appendChild(canvas);

        // If you want to convert it back to an image URL
        const croppedImageURL = canvas.toDataURL('image/jpeg');
        console.log(croppedImageURL);
    };
}

// Example usage:
const inputImagePath = 'path/to/your/input/image.jpg';
cropToSquare(inputImagePath);
